modules/security-aai/src/main/java/org/opencastproject/security/aai/ConfigurableLoginHandler.java

```diff
@@ -425,6 +465,14 @@ private String extractEmail(HttpServletRequest request) {
     if (StringUtils.equals(id, bootstrapUserId)) {
       roles.add(new JpaRole(GLOBAL_ADMIN_ROLE, organization));
     }
+	if (headerAffiliation != null) {
+      String affiliation = request.getHeader(headerAffiliation);
```

Comment:
Depending on the Shibboleth federation and the configuration of the Shibboleth service provider, it might be possible that this header is not set at all.

Therefore, you need to add the check "if (affiliation != null)".

Reply:
Since this feature is optional, only if on the configuration file "header.affiliation" is uncommented and set to the correct value, this feature will work, therefore no check for "affiliation" value is necessary.

Reply:
In case of a faulty configuration/typo by system administrator "affiliation" check added, as a safety measure.